ghc-lib-gen/src/Main.hs

```diff
@@ -65,11 +65,14 @@ ghcLibParserIncludeDirs =
 -- adjusting.
 ghcLibParserHsSrcDirs :: [Cabal] -> [FilePath]
 ghcLibParserHsSrcDirs lib =
-  nubSort $
-   [ "ghc-lib/stage0/compiler/build"
-   , "ghc-lib/stage1/compiler/build"] ++
-   map takeDirectory cabalFileLibraries ++
-   askFiles lib "hs-source-dirs:"
+  -- Sort by length so the longest paths are at the front.
```

Comment:
This comment tells me what we are doing but not why this is necessary.

Reply:
✔️